src/main/java/org/sonarsource/sonarlint/ls/AnalysisTaskExecutor.java

```diff
@@ -313,8 +323,11 @@ private static boolean isPrefixForAll(Path prefixCandidate, Collection<Path> pat
   }
 
   private void analyzeSingleModuleNonExcluded(AnalysisTask task, WorkspaceFolderSettings settings, Optional<ProjectBindingWrapper> binding,
-    Map<URI, VersionedOpenFile> filesToAnalyze, URI baseDirUri, Map<URI, GetJavaConfigResponse> javaConfigs) {
+    Map<URI, VersionedOpenFile> filesToAnalyze, URI baseDirUri, Map<URI, GetJavaConfigResponse> javaConfigs, @Nullable ProgressFacade progressFacade) {
     task.checkCanceled();
+    if (progressFacade != null) {
+      progressFacade.checkCanceled();
+    }
     if (filesToAnalyze.size() == 1) {
```

Comment:
This if-else could be extracted to method like `logAnalysis()` to reduce the code complexity here. Up to you.